tests/5.2/declare_mapper/test_declare_mapper_present.c

```diff
@@ -0,0 +1,67 @@
+//===---- test_declare_mapper_present.c -------------------------------------===//
+// 
+// OpenMP API Version 5.2 
+//
+// This example has been adapted from the 5.2 OpenMP Examples document:
+// "declare mapper Directive", and "OpenMP Directive Syntax -> complex iterator"
+// The declare mapper directive will be used to automatically map variables
+// according to its prescription:  full structure, plus the dynamic storage of the
+// data element. These variable should be present on the device.
+//
+//===-------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1000
+
+typedef struct myvec{
+    size_t len;
+    double *data;
+} myvec_t;
+
+
+#pragma omp declare mapper(default:myvec_t v) map(present, tofrom: v, v.data[0:v.len]) 
+
+void init( myvec_t *s )
+{ 
+  for(size_t i = 0; i < s->len; i++)
+    s->data[i] = i; 
+}
+
+
+int test_declare_mapper_present() { 
+
+  OMPVV_INFOMSG("test_declare_mapper_present");
+  int errors = 0;
+
+   myvec_t s;
+
+   s.data = (double *)calloc(N,sizeof(double));
+   s.len  = N;
+  #pragma omp target data map(tofrom: s, s.data[0:s.len])
+  {
+    #pragma omp target
+    {
+        init(&s);
+    }
+  }
```

Comment:
```suggestion
  #pragma omp target map(s)
  {
      init(&s);
  }

```
As the test is written, I consider that it is not testing that the `declare mapper` is working. It is only testing if the compiler accepts the directive is recognized by the compiler. This is because the mapper is not used. With this modification the mapper is used.

Reply:
I guess that the reason for having the nesting `target data` construct is to test the `present` modifier in line 26.
If we remove the nesting `target data` construct, the compiled program will fail at runtime due to the `present` requirement.

Reply:
I would agree with the suggested change as it relies on the declare mapper characteristics and does not influence how s is mapped.